src/seals/base_envs.py

```diff
@@ -97,32 +116,58 @@ def seed(self, seed=None) -> Sequence[int]:
 
     def reset(self) -> Observation:
         """Reset episode and return initial observation."""
-        self.cur_state = self.initial_state()
-        assert self.cur_state in self.state_space, f"unexpected state {self.cur_state}"
+        self.state = self.initial_state()
         self._n_actions_taken = 0
-        return self.obs_from_state(self.cur_state)
+        return self.obs_from_state(self.state)
 
     def step(self, action: Action) -> Tuple[Observation, float, bool, dict]:
         """Transition state using given action."""
-        if self.cur_state is None or self._n_actions_taken is None:
+        if self._cur_state is None or self._n_actions_taken is None:
             raise ValueError("Need to call reset() before first step()")
         if action not in self.action_space:
             raise ValueError(f"{action} not in {self.action_space}")
 
-        old_state = self.cur_state
-        self.cur_state = self.transition(self.cur_state, action)
-        assert self.cur_state in self.state_space, f"unexpected state {self.cur_state}"
-        obs = self.obs_from_state(self.cur_state)
-        assert obs in self.observation_space, f"{obs} not in {self.observation_space}"
-        rew = self.reward(old_state, action, self.cur_state)
-        done = self.terminal(self.cur_state, self._n_actions_taken)
+        old_state = self.state
+        self.state = self.transition(self.state, action)
+        obs = self.obs_from_state(self.state)
+        if obs not in self.observation_space:
+            raise ValueError(f"{obs} not in {self.observation_space}")
+        reward = self.reward(old_state, action, self.state)
         self._n_actions_taken += 1
+        done = self.terminal(self.state, self.n_actions_taken)
+
+        infos = {"old_state": old_state, "new_state": self._cur_state}
+        return obs, reward, done, infos
+
+
+class ExposePOMDPStateWrapper(gym.Wrapper, Generic[State, Observation, Action]):
+    """A wrapper that exposes the current state of the POMDP as the observation."""
+
+    def __init__(self, env: ResettablePOMDP[State, Observation, Action]) -> None:
+        """Build wrapper.
+
+        Args:
+            env: POMDP to wrap.
+        """
+        super().__init__(env)
+        self._observation_space = env.state_space
+
+    def reset(self) -> State:
+        """Reset environment and return initial state."""
+        self.env.reset()
+        return self.env.state
 
-        infos = {"old_state": old_state, "new_state": self.cur_state}
-        return obs, rew, done, infos
+    def step(self, action) -> Tuple[State, float, bool, dict]:
+        """Transition state using given action."""
+        obs, reward, done, info = self.env.step(action)
+        return self.env.state, reward, done, info
 
 
-class ResettableMDP(ResettablePOMDP[State, State, Action], Generic[State, Action]):
+class ResettableMDP(
+    ResettablePOMDP[State, State, Action],
+    abc.ABC,
```

Comment:
I get your point. From the Python standard/implementation point of view, this is indeed unnecessary; a class becomes concrete when there are no more abstract methods left to implement. I think however it is a style choice that I've seen around that if the subclass is still abstract, you should add an explicit ABC dependency to make code self-documenting -- it might be hard to visually check if *all* or only 90% of the methods of a large class have been implemented. (cannot find a reference for this now, but PyCharm encourages enforcing this style).

Reply:
I think I kind of agree with this style choice, even though here it's more obvious that most methods are unimplemented, but I don't really mind as long as we're internally consistent.